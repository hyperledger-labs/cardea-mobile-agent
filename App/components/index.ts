// DEPRECATED: This creates dependency cycles

import Button from './buttons/Button'
import CheckBoxRow from './inputs/CheckBoxRow'
import QRScanner from './inputs/QRScanner'
import TextInput from './inputs/TextInput'
import ContactListItem from './listItems/ContactListItem'
import CredentialListItem from './listItems/CredentialListItem'
import NotificationCredentialListItem from './listItems/NotificationCredentialListItem'
import NotificationProofListItem from './listItems/NotificationProofListItem'
import HighlightTextBox from './texts/HighlightTextBox'
import InfoTextBox from './texts/InfoTextBox'
import Label from './texts/Label'
import Text from './texts/Text'
import Title from './texts/Title'
import ModularView from './views/ModularView'
import SafeAreaScrollView from './views/SafeAreaScrollView'

export {
  CheckBoxRow,
  ContactListItem,
  CredentialListItem,
  Label,
  ModularView,
  Button,
  NotificationCredentialListItem,
  NotificationProofListItem,
  QRScanner,
  SafeAreaScrollView,
  TextInput,
  Text,
  Title,
  InfoTextBox,
  HighlightTextBox,
}
