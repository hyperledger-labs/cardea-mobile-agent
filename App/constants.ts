import { IndyCredentialMetadata } from '@aries-framework/core/build/types'

export const defaultLanguage = 'en'

export enum LocalStorageKeys {
  Onboarding = 'OnboardingState',
}

export enum Screens {
  Onboarding = 'Onboarding',
  Terms = 'Terms',
  CreatePin = 'Create 6-Digit Pin',
  Splash = 'Splash',
  EnterPin = 'Enter Pin',
}

// FIXME: Remove once fixed in AFJ
export interface IndexedIndyCredentialMetadata extends IndyCredentialMetadata {
  [key: string]: string | undefined
}

export const indyCredentialKey = '_internal/indyCredential'

export const dateFormatString = 'LLL d, yyyy'
export const dateTimeFormatString = 'LLL d, yyyy (hh:mm a)'
// DEPRECATED
export const credentialDateTimeFormatString = 'LLL d, yyyy (hh:mm a)'
