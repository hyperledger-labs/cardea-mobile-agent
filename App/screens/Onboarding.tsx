import { useFocusEffect } from '@react-navigation/native'
import React, { Ref, useCallback, useRef, useState } from 'react'
import { Animated, BackHandler, Dimensions, FlatList, View } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'

import { Pagination } from '../components/shared/Pagination'

const { width } = Dimensions.get('window')

export interface OnboardingStyleSheet {
  container: Record<string, any>
  carouselContainer: Record<string, any>
  pagerContainer: Record<string, any>
  pagerDot: Record<string, any>
  pagerPosition: Record<string, any>
  pagerNavigationButton: Record<string, any>
}

interface OnboardingProps {
  pages: Array<Element>
  nextButtonText: string
  previousButtonText: string
  style: OnboardingStyleSheet
}

const Onboarding: React.FC<OnboardingProps> = ({ pages, nextButtonText, previousButtonText, style }) => {
  const [activeIndex, setActiveIndex] = useState(0)
  const flatList: Ref<FlatList> = useRef(null)
  const scrollX = useRef(new Animated.Value(0)).current

  const onViewableItemsChangedRef = useRef(({ viewableItems }: any) => {
    if (!viewableItems[0]) {
      return
    }

    setActiveIndex(viewableItems[0].index)
  })

  const viewabilityConfigRef = useRef({
    viewAreaCoveragePercentThreshold: 60,
  })

  const onScroll = Animated.event([{ nativeEvent: { contentOffset: { x: scrollX } } }], {
    useNativeDriver: false,
  })

  const next = () => {
    if (activeIndex + 1 < pages.length) {
      flatList?.current?.scrollToIndex({
        index: activeIndex + 1,
        animated: true,
      })
    }
  }

  const previous = () => {
    if (activeIndex !== 0) {
      flatList?.current?.scrollToIndex({
        index: activeIndex - 1,
        animated: true,
      })
    }
  }

  const renderItem = useCallback(
    ({ item, index }: { item: Element; index: number }) => (
      <View key={index} style={[{ width }, style.carouselContainer]}>
        {item}
      </View>
    ),
    []
  )

  useFocusEffect(
    useCallback(() => {
      const onBackPress = () => {
        BackHandler.exitApp()
        return true
      }
      BackHandler.addEventListener('hardwareBackPress', onBackPress)
      return () => BackHandler.removeEventListener('hardwareBackPress', onBackPress)
    }, [])
  )

  return (
    <SafeAreaView style={style.container}>
      <FlatList
        ref={flatList}
        horizontal
        pagingEnabled
        showsHorizontalScrollIndicator={false}
        style={[{ width }]}
        data={pages}
        renderItem={renderItem}
        viewabilityConfig={viewabilityConfigRef.current}
        onViewableItemsChanged={onViewableItemsChangedRef.current}
        onScroll={onScroll}
        scrollEventThrottle={16}
      />
      <Pagination
        pages={pages}
        activeIndex={activeIndex}
        nextButtonText={nextButtonText}
        previousButtonText={previousButtonText}
        scrollX={scrollX}
        style={style}
        next={next}
        previous={previous}
      />
    </SafeAreaView>
  )
}

export default Onboarding
