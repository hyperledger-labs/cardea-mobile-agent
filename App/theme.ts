import {StyleSheet} from 'react-native'

type BaseColor = Record<string, string>

export const borderRadius = 4
export const heavyOpacity = 0.7
export const lightOpacity = 0.35
export const zeroOpacity = 0.0

export const BaseColors: BaseColor = {
  black: '#000000',
  darkBlue: '#003366',
  darkBlueLightTransparent: `rgba(0, 51, 102, ${heavyOpacity})`,
  darkBlueHeavyTransparent: `rgba(0, 51, 102, ${lightOpacity})`,
  darkGreen: '#35823F',
  darkGreenLightTransparent: `rgba(53, 130, 63, ${heavyOpacity})`,
  darkGreenHeavyTransparent: `rgba(53, 130, 63, ${lightOpacity})`,
  darkOrange: '#FAA220',
  darkOrangeLightTransparent: `rgba(250, 162, 32, ${heavyOpacity})`,
  darkOrangeHeavyTransparent: `rgba(250, 162, 32, ${lightOpacity})`,
  darkGrey: '#1C1C1E',
  darkGreyAlt: '#313132',
  green: '#2D6E35',
  lightBlue: '#759ECD',
  lightGrey: '#D3D3D3',
  lightGreyAlt: '#F2F2F2',
  mediumBlue: '#B9CEDE',
  offWhite: '#F2F2F2',
  red: '#DE3333',
  transparent: `rgba(0, 0, 0, ${zeroOpacity})`,
  yellow: '#FCBA19',
  white: '#FFFFFF',
}

export const StatusColors: BaseColor = {
  error: BaseColors.red,
  info: BaseColors.black,
  success: BaseColors.green,
  warning: BaseColors.black,
}

interface ColorTheme extends BaseColor {
  accent: string
  background: string
  backgroundLight: string
  borderLight: string
  primary: string
  primaryActive: string
  shadow: string
  text: string
}

interface FontAttributes {
  fontSize: number
  fontWeight:
    | 'normal'
    | 'bold'
    | '100'
    | '200'
    | '300'
    | '400'
    | '500'
    | '600'
    | '700'
    | '800'
    | '900'
  color: string
}

interface TextTheme {
  headingOne: FontAttributes
  headingTwo: FontAttributes
  headingThree: FontAttributes
  headingFour: FontAttributes
  normal: FontAttributes
  label: FontAttributes
  caption: FontAttributes
}

interface CredentialTheme {
  background: string
}

interface CredentialOfferTheme {
  background: string
}

interface ContactTheme {
  background: string
}

interface ModularViewTheme {
  background: string
}

interface ProofRequestTheme {
  background: string
}

interface SettingsTheme {
  background: string
}

interface SingleSelectBlockTheme {
  background: string
}

interface TextBoxTheme {
  background: string
  border: string
  text: string
}

export const Colors: ColorTheme = {
  accent: BaseColors.lightBlue,
  background: BaseColors.white,
  backgroundLight: BaseColors.lightGrey,
  borderLight: BaseColors.darkOrangeLightTransparent,
  primary: BaseColors.darkOrange,
  primaryActive: BaseColors.darkOrangeHeavyTransparent,
  shadow: BaseColors.darkGrey,
  text: BaseColors.black,
  ...BaseColors,
  ...StatusColors,
}

export const CredentialTheme: CredentialTheme = {
  background: Colors.offWhite,
}

export const CredentialOfferTheme: CredentialOfferTheme = {
  background: Colors.lightGrey,
}

export const ContactTheme: ContactTheme = {
  background: Colors.offWhite,
}

export const ModularViewTheme: ModularViewTheme = {
  background: Colors.offWhite,
}

export const ProofRequestTheme: ProofRequestTheme = {
  background: Colors.lightGrey,
}

export const SettingsTheme: SettingsTheme = {
  background: Colors.offWhite,
}

export const SingleSelectBlockTheme: SingleSelectBlockTheme = {
  background: Colors.offWhite,
}

export const TextBoxTheme: TextBoxTheme = {
  background: Colors.darkOrangeLightTransparent,
  border: Colors.borderLight,
  text: Colors.text,
}

export const TextTheme: TextTheme = {
  headingOne: {
    fontSize: 38,
    fontWeight: 'bold',
    color: Colors.text,
  },
  headingTwo: {
    fontSize: 32,
    fontWeight: 'bold',
    color: Colors.text,
  },
  headingThree: {
    fontSize: 26,
    fontWeight: 'bold',
    color: Colors.text,
  },
  headingFour: {
    fontSize: 21,
    fontWeight: 'bold',
    color: Colors.text,
  },
  normal: {
    fontSize: 18,
    fontWeight: 'normal',
    color: Colors.text,
  },
  label: {
    fontSize: 14,
    fontWeight: 'bold',
    color: Colors.text,
  },
  caption: {
    fontSize: 14,
    fontWeight: 'normal',
    color: Colors.text,
  },
}

export const Buttons = StyleSheet.create({
  primary: {
    padding: 16,
    borderRadius: 4,
    backgroundColor: Colors.primary,
  },
  primaryDisabled: {
    padding: 16,
    borderRadius: 4,
    backgroundColor: Colors.darkOrangeHeavyTransparent,
  },
  primaryText: {
    ...TextTheme.normal,
    fontWeight: 'bold',
    color: Colors.white,
    textAlign: 'center',
  },
  primaryTextDisabled: {
    ...TextTheme.normal,
    fontWeight: 'bold',
    color: Colors.white,
    textAlign: 'center',
  },
  secondary: {
    padding: 16,
    borderRadius: 4,
    borderWidth: 2,
    borderColor: Colors.primary,
    backgroundColor: Colors.white,
  },
  secondaryDisabled: {
    padding: 16,
    borderRadius: 4,
    borderWidth: 2,
    borderColor: Colors.darkOrangeLightTransparent,
    backgroundColor: Colors.white,
  },
  secondaryText: {
    ...TextTheme.normal,
    fontWeight: 'bold',
    color: Colors.blue,
    textAlign: 'center',
  },
  secondaryTextDisabled: {
    ...TextTheme.normal,
    fontWeight: 'bold',
    color: Colors.lightBlue,
    textAlign: 'center',
  },
})
