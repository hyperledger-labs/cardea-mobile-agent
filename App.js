/**
 * Cardea Mobile App - Holder
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, {useEffect, useState} from 'react'
import {View} from 'react-native'
import {Redirect, Route} from 'react-router-native'

import 'react-native-get-random-values'
import '@azure/core-asynciterator-polyfill'

import * as Keychain from 'react-native-keychain'
import AsyncStorage from '@react-native-community/async-storage'
import {AgentProvider} from './components/AgentProvider/'
import Errors from './components/Errors/index.js'
import Notifications from './components/Notifications/index.js'
import AddressInfo from './components/Registration/AddressInfo/index.js'
import Confirmed from './components/Registration/Confirmed/index.js'
import ConnectNow from './components/Registration/ConnectNow/index.js'
import ContactInfo from './components/Registration/ContactInfo/index.js'
import EntryPoint from './components/EntryPoint/index.js'
import Home from './components/Home/index.js'
import ListContacts from './components/ListContacts/index.js'
import ListCredentials from './components/ListCredentials/index.js'
import Navbar from './components/Navbar/index.js'
import PinCreate from './components/Registration/PinCreate/index.js'
import PinEnter from './components/PinEnter/index.js'
import Settings from './components/Settings/index.js'
import SetupWizard from './components/Registration/SetupWizard/index.js'
import StartScreen from './components/StartScreen/index.js'
import Terms from './components/Registration/Terms/index.js'
import Workflow from './components/Workflow/index.js'
import Passport from './components/Registration/Passport/index.js'
import {PrivacyPolicy} from '@configs/policyConfigs'
import PassportDirections from './components/Registration/PassportDirections'
import AppStyles from '@assets/styles'

const App = (props) => {
  const [authenticated, setAuthenticated] = useState(false)
  const [connection, setConnection] = useState(undefined)
  const [credential, setCredential] = useState(undefined)

  useEffect(() => {
    const clearKeychain = async () => {
      try {
        const value = await AsyncStorage.getItem('firstLaunch')
        if (value == null) {
          // Detecting a first time launch, resetting keychain
          console.log('First launch. Clearing keychain.')
          let resetArray = [
            'passcode',
            'setupWizard',
            'governmentConnectionId',
            'demographicData',
            'passportData',
            'trustedTraveler',
          ]
          const resetVal = async (val) => {
            await Keychain.resetGenericPassword({service: val})
          }
          resetArray.forEach((val) => resetVal(val))
          await AsyncStorage.setItem('firstLaunch', 'false')
        }
      } catch (e) {
        console.warn(e)
      }
    }
    clearKeychain()
  }, [])

  /*
    /
    /home
    /start
    /pin-entry
    /pin-create
    /scan
    /workflow
    /settings
  */

  return (
    <View style={{flex: 1}}>
      <AgentProvider>
        <Errors>
          <Notifications>
            <View style={{flex: 1}}>
              <Route
                exact
                path="/"
                render={() => {
                  return <EntryPoint authenticated={authenticated} />
                }}
              />
              <Route
                exact
                path="/home"
                render={() =>
                  authenticated ? (
                    <Home
                      connection={connection}
                      setConnection={setConnection}
                      credential={credential}
                      setCredential={setCredential}
                    />
                  ) : (
                    <Redirect to="/" />
                  )
                }
              />
              <Route
                exact
                path="/settings"
                render={() =>
                  authenticated ? <Settings /> : <Redirect to="/" />
                }
              />
              <Route
                exact
                path="/contacts"
                render={() =>
                  authenticated ? <ListContacts /> : <Redirect to="/" />
                }
              />
              <Route
                exact
                path="/credentials"
                render={() =>
                  authenticated ? <ListCredentials /> : <Redirect to="/" />
                }
              />
              <Route
                path="/workflow"
                render={() => (
                  <Workflow
                    connection={connection}
                    setConnection={setConnection}
                    credential={credential}
                    setCredential={setCredential}
                    authenticated={authenticated}
                  />
                )}
              />
              <Route
                exact
                path="/pin/enter"
                render={() => <PinEnter setAuthenticated={setAuthenticated} />}
              />
              <Route exact path="/start" render={() => <StartScreen />} />
              <Route
                path="/setup-wizard"
                render={() => (
                  <SetupWizard setAuthenticated={setAuthenticated}>
                    <Terms
                      title={PrivacyPolicy.title}
                      message={PrivacyPolicy.text}
                    />
                    <PinCreate />
                    <PassportDirections />
                    <Passport />
                    <ContactInfo />
                    <AddressInfo />
                    <ConnectNow />
                    <Confirmed />
                  </SetupWizard>
                )}
              />
            </View>
            {authenticated ? (
              <View style={{height: '10.5%'}}>
                <Navbar />
              </View>
            ) : null}
          </Notifications>
        </Errors>
      </AgentProvider>
    </View>
  )
}

export default App
