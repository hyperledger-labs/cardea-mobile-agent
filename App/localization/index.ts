import AsyncStorage from '@react-native-async-storage/async-storage'
import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import * as RNLocalize from 'react-native-localize'

import { defaultLanguage } from '../constants'

import en from './en'
import fr from './fr'

export type Translation = typeof en

const resources = {
  en: {
    translation: en,
  },
  fr: {
    translation: fr,
  },
}

export enum Locales {
  en = 'en',
  fr = 'fr',
}

const currentLanguage = i18n.language

//** Store language into the AsyncStorage  */
const storeLanguage = async (id: string) => {
  await AsyncStorage.setItem('language', id)
}

//** Fetch user preference language from the AsyncStorage and set if require  */
const initStoredLanguage = async () => {
  const langId = await AsyncStorage.getItem('language')
  if (langId !== null) {
    if (langId !== currentLanguage) {
      await i18n.changeLanguage(langId)
    }
  }
}

const availableLanguages = Object.keys(resources)
const bestLanguageMatch = RNLocalize.findBestAvailableLanguage(availableLanguages)
let translationToUse = defaultLanguage

if (bestLanguageMatch && availableLanguages.includes(bestLanguageMatch.languageTag)) {
  translationToUse = bestLanguageMatch.languageTag
}

i18n.use(initReactI18next).init({
  debug: true,
  lng: translationToUse,
  fallbackLng: defaultLanguage,
  resources,
})

export { i18n, initStoredLanguage, storeLanguage, currentLanguage }
