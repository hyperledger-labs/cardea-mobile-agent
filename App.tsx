import {
  Agent,
  AutoAcceptCredential,
  ConsoleLogger,
  HttpOutboundTransport,
  LogLevel,
  MediatorPickupStrategy,
  WsOutboundTransport,
} from '@aries-framework/core'
import AgentProvider from '@aries-framework/react-hooks'
import { agentDependencies } from '@aries-framework/react-native'
import { default as React, useEffect, useState } from 'react'
import { StatusBar } from 'react-native'
import Config from 'react-native-config'
import SplashScreen from 'react-native-splash-screen'
import Toast from 'react-native-toast-message'

import toastConfig from './App/components/toast/ToastConfig'
import { initStoredLanguage } from './App/localization'
import RootStack from './App/navigators/RootStack'
import StoreProvider from './App/store/Store'
import { Colors } from './App/theme'
import indyLedgers from './configs/ledgers/indy'
console.disableYellowBox = true

const App = () => {
  const [agent, setAgent] = useState<Agent | undefined>(undefined)

  initStoredLanguage()

  const initAgent = async () => {
    const newAgent = new Agent(
      {
        label: 'Cardea Holder',
        mediatorConnectionsInvite: Config.MEDIATOR_URL,
        mediatorPickupStrategy: MediatorPickupStrategy.Implicit,
        walletConfig: { id: 'wallet4', key: '123' },
        autoAcceptConnections: true,
        autoAcceptCredentials: AutoAcceptCredential.ContentApproved,
        logger: new ConsoleLogger(LogLevel.trace),
        indyLedgers,
      },
      agentDependencies
    )

    const wsTransport = new WsOutboundTransport()
    const httpTransport = new HttpOutboundTransport()

    newAgent.registerOutboundTransport(wsTransport)
    newAgent.registerOutboundTransport(httpTransport)

    await newAgent.initialize()
    setAgent(newAgent)
  }

  useEffect(() => {
    // Hide the native splash / loading screen so that our
    // RN version can be displayed.
    SplashScreen.hide()
    initAgent()
  }, [])

  return (
    <StoreProvider>
      <AgentProvider agent={agent}>
        <StatusBar barStyle="light-content" hidden={false} backgroundColor={Colors.primary} translucent={false} />
        <RootStack />
        <Toast topOffset={15} config={toastConfig} />
      </AgentProvider>
    </StoreProvider>
  )
}

export default App
