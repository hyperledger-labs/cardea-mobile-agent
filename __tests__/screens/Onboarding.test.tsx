import React from 'react'
import { StyleSheet, Text } from 'react-native'
import renderer from 'react-test-renderer'

import Onboarding, { OnboardingStyleSheet } from '../../App/screens/Onboarding'
import { Colors } from '../../App/theme'

const markTutorialFin = jest.fn()

export const carousel: OnboardingStyleSheet = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: Colors.background,
  },
  carouselContainer: {
    flexDirection: 'column',
    backgroundColor: Colors.background,
  },
  pagerContainer: {
    flexShrink: 2,
    flexDirection: 'row',
    alignItems: 'center',
  },
  pagerDot: {
    borderWidth: 1,
    borderStyle: 'solid',
    borderColor: Colors.primary,
  },
  pagerPosition: {
    position: 'relative',
    top: 0,
  },
  pagerNavigationButton: {
    fontSize: 18,
    fontWeight: 'bold',
    color: Colors.primary,
  },
})

const pages = [
  <>
    <Text testID={'bodyText'}>Hello</Text>
  </>,
  <>
    <Text testID={'bodyText'}>World</Text>
  </>,
]

jest.mock('@react-navigation/native', () => {
  return {
    useFocusEffect: jest.fn(),
  }
})

describe('Onboarding', () => {
  it('Renders correctly', () => {
    const tree = renderer
      // @ts-ignore
      .create(<Onboarding pages={pages} onOnboardingDismissed={markTutorialFin} style={carousel} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  // it('Next works correctly', () => {
  //   // @ts-ignore
  //   const tree = renderer.create(<Onboarding pages={pages} onOnboardingDismissed={markTutorialFin} style={carousel} />)
  //   const b = tree.root!.findByProps({ testID: 'nextButton' })

  //   expect(markTutorialFin).toBeCalledTimes(1)
  // })

  it('Pages exist', () => {
    // @ts-ignore
    const tree = renderer.create(<Onboarding pages={pages} onOnboardingDismissed={markTutorialFin} style={carousel} />)
    const foundPages = tree.root!.findAllByType(Text).filter((e: any) => e.props.testID === 'bodyText')

    expect(foundPages.length).toBe(2)
  })
})
