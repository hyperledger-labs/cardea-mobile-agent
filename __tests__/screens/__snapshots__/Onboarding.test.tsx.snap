// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding Renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        <React.Fragment>
          <Text
            testID="bodyText"
          >
            Hello
          </Text>
        </React.Fragment>,
        <React.Fragment>
          <Text
            testID="bodyText"
          >
            World
          </Text>
        </React.Fragment>,
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={16}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "width": 750,
        },
      ]
    }
    viewabilityConfig={
      Object {
        "viewAreaCoveragePercentThreshold": 60,
      }
    }
    viewabilityConfigCallbackPairs={
      Array [
        Object {
          "onViewableItemsChanged": [Function],
          "viewabilityConfig": Object {
            "viewAreaCoveragePercentThreshold": 60,
          },
        },
      ]
    }
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "width": 750,
              },
              Object {
                "backgroundColor": "#000000",
                "flexDirection": "column",
              },
            ]
          }
        >
          <Text
            testID="bodyText"
          >
            Hello
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "width": 750,
              },
              Object {
                "backgroundColor": "#000000",
                "flexDirection": "column",
              },
            ]
          }
        >
          <Text
            testID="bodyText"
          >
            World
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 2,
      }
    }
  >
    <View
      accessibilityLabel="Previous"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="previousButton"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#35823F",
              "fontSize": 18,
              "fontWeight": "bold",
            },
            Object {
              "color": "transparent",
              "paddingRight": 20,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "bottom": 20,
            "flexDirection": "row",
            "position": "absolute",
          },
          Object {
            "position": "relative",
            "top": 0,
          },
        ]
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "backgroundColor": "rgba(53, 130, 63, 1)",
            "borderColor": "#35823F",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 10,
            "marginHorizontal": 5,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": 10,
          }
        }
      />
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "borderColor": "#35823F",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 10,
            "marginHorizontal": 5,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="Next"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="nextButton"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#35823F",
              "fontSize": 18,
              "fontWeight": "bold",
            },
            Object {
              "color": "#35823F",
              "paddingLeft": 20,
            },
          ]
        }
      />
    </View>
  </View>
</RNCSafeAreaView>
`;
